### DISCUSSION TITLE
Solving Port ID Encoding Issue for ICA (ICS-27) opening the 'icacontroller-' Channel

### DISCUSSION THREAD

**Discussion by magiodev** [informalsystems/hermes#3177] [Show and tell]
Hello folks,

I recently encountered an issue while using Hermes relayer to open a channel for ICA (ICS-27). The problem was related to encoding the Port ID. I believe the solution could be useful to others, so I wanted to share it here.

Issue:
When opening a channel for ICA (ICS-27) using Hermes relayer, I faced an issue with encoding the Port ID. I initially passed the Port ID as 'ics-27', but it caused an error.

Solution:
To resolve this issue, I modified the way I passed the Port ID. Instead of just 'ics-27', I passed it as a JSON string containing more information about the channel:
```json
'{"version":"ics27-1","encoding":"proto3","tx_type":"sdk_multi_msg","controller_connection_id":"connection-0","host_connection_id":"connection-0"}'
```
After passing the Port ID this way, I was able to successfully open the channel for ICA (ICS-27) without any issues.

I hope this information helps others who might be facing similar problems. Please feel free to ask questions or share your thoughts on this solution.

Cheers!

**Comment by romac** [comment]
For ICA, the port identifiers must be specified as follows:

> Controller Port Identifier: `optional prefix icacontroller- + mandatory {owner-account-address}`
> 
> Host Port Identifier: `icahost`
> 
> The `icacontroller-` prefix on the controller port identifier is optional and host chains must not enforce that the counterparty port identifier includes it. Controller chains may decide to include it and validate that it is present in their own port identifier.

https://github.com/cosmos/ibc/blob/main/spec/app/ics-027-interchain-accounts/README.md#identifier-formats

On the other hand, as you found out, it is the version which holds the metadata about the channel, and which must be specified as a JSON-encoded object: https://github.com/cosmos/ibc/blob/main/spec/app/ics-027-interchain-accounts/README.md#metadata-negotiation-summary

Hope that helps!
